src/DeployPage.js

```diff
@@ -197,22 +197,40 @@ class DeployTable extends React.Component {
       PropTypes.shape({ name: PropTypes.string.isRequired })
     ).isRequired,
     commits: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
-    tags: PropTypes.object.isRequired
+    tags: PropTypes.object.isRequired,
+    code: PropTypes.string.isRequired
   };
+  static prefBorsModeCacheKey = 'pref-bors-mode';
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      borsMode: false
-    };
-  }
+  getDefaultBorsMode = () => {
+    const prefs = JSON.parse(
+      localStorage.getItem(this.prefBorsModeCacheKey) || '{}'
+    );
+    return !!prefs[this.props.code];
+  };
+
+  state = {
+    borsMode: this.getDefaultBorsMode()
```

Comment:
I don't think this will correctly handle if the props of the page change, as it won't re-check for the new code. I think it needs to be handled like we do fetching remote content instead.

Reply:
True. I'll make it a lifecycle hook instead.